src/content.ts

```diff
@@ -2,36 +2,111 @@ if (document.location.href === 'https://en.wikipedia.org/wiki/Pug') {
-            const wordStart = textContent.indexOf(word);
-            if (wordStart > -1) {
-                const beforeWord = textContent.slice(0, wordStart);
-                const afterWord = textContent.slice(wordStart + word.length);
+const CHECK_ONLY_SPACES_REGEXP = /^\s*$/
+const taggedWordElementClassName = 'tagged-word';
+
+function parsePage(node) {
+    let textContent = "";
+    const textNodes = {};
+    
+    
+    processNode(node);
 
-                node.textContent = beforeWord;
-                const wordElement = document.createElement('span');
-                wordElement.innerText = `${word} (${translation})`;
+    return {
+        textContent,
+        textNodes
+    };
 
-                const afterWordElement = document.createTextNode(afterWord);
-                node.after(wordElement,afterWordElement);
+    function processNode(node) {
 
+        if(node.nodeType === 3){
+            if(node.data.match(CHECK_ONLY_SPACES_REGEXP)){
+                return;
             }
-        }
-        else {
-            console.log('Children of', node);
-            find(word, translation, node);
+            textNodes[textContent.length] = node;
+            textContent += node.data;
+            return;
+        } 
+        
+        let childNodes = Array.from(node.childNodes || []);
+    
+        for(const childNode of childNodes) {
+            processNode(childNode)
         }
     }
 }
 
+const {textContent, textNodes}  = parsePage(document.body);
+
+console.log("All Text Nodes:", textContent, textNodes);
+
+function searchWord(textNode, nodeOffset, wordOffset, wordLength) {
+    
+    const taggedWord = textNode.textContent.substr(wordOffset, wordLength); //получаем нужное слово
+    const endText = textNode.textContent.slice(wordOffset+wordLength); //все что после нужного слова
+    const taggedWordElement = document.createElement("a");
+
+    taggedWordElement.textContent = taggedWord;
+    taggedWordElement.className = taggedWordElementClassName; 
+    textNode.textContent = textNode.textContent.substr(0, wordOffset);
+    textNode.after(taggedWordElement, endText);
+    
+    textNodes[nodeOffset+textNode.length] = taggedWordElement.childNodes[0];
+    textNodes[nodeOffset+textNode.length+taggedWord.length] = taggedWordElement.nextSibling;
+}
+
+searchWord(textNodes["290"], 290, 7, 4);
+console.log(textNodes["290"])
+
+
+function createTooltip(textTooltip) {
+    const tooltipElement = document.createElement("a");
+    tooltipElement.className = 'tooltip';
+    tooltipElement.textContent = textTooltip;
+    
+    return tooltipElement
+}
+
+const tooltipElement = createTooltip("temporary text");
+
+document.body.append(tooltipElement);
+
+const style = document.querySelector('style');
+
+function addStyle(styleString) {  
+    style.textContent = styleString;
+}
+
+document.addEventListener('mouseover', function(event) {
+    if(event.target.classList.contains(taggedWordElementClassName)) {
+        addStyle(`
```

Comment:
"addStyle" should be called only once when code is executed. No need to add styles each time mouseover happens.